node/node.go

```diff
@@ -886,6 +886,7 @@ func NewNode(config *cfg.Config,
 		mempool,
 		evidencePool,
 		sm.BlockExecutorWithMetrics(smMetrics),
+		sm.WithBlockStore(blockStore),
```

Comment:
I see that we're also add this option to tests. can we still run without the blockstore? I see we have the `if blockstore == nil` statement, which is good, but want to double check if this is a desired goal. if so, do we have a test? if not, should we be passing this an option if its not optional?

Reply:
yes, blockstore is optional and TestApplyBlock initialized block executor without the optional arguments https://github.com/celestiaorg/celestia-core/blob/ff2bff4a29aec6851499e34b7d84349d8e774dd9/state/execution_test.go#L47